app/src/androidTest/java/com/github/se/polyfit/ui/screen/IngredientTest.kt

```diff
@@ -0,0 +1,175 @@
+package com.github.se.polyfit.ui.screen
+
+import androidx.compose.ui.test.assertCountEquals
+import androidx.compose.ui.test.assertIsDisplayed
+import androidx.compose.ui.test.junit4.createComposeRule
+import androidx.compose.ui.test.onAllNodesWithTag
+import androidx.navigation.compose.rememberNavController
+import androidx.test.ext.junit.runners.AndroidJUnit4
+import com.github.se.polyfit.ui.navigation.Navigation
+import com.kaspersky.kaspresso.testcases.api.testcase.TestCase
+import io.github.kakaocup.compose.node.element.ComposeScreen
+import org.junit.Rule
+import org.junit.Test
+import org.junit.runner.RunWith
+
+@RunWith(AndroidJUnit4::class)
+class IngredientTest : TestCase() {
+  @get:Rule val composeTestRule = createComposeRule()
+
+  private fun launchIngredientScreenWithTestData(
+      testIngredients: List<Ingredient>,
+      testPotentials: List<Ingredient>
+  ) {
+    composeTestRule.setContent {
+      val navController = rememberNavController()
+      IngredientScreen(Navigation(navController), testIngredients, testPotentials)
+    }
+  }
+
+  private val manyIngredients =
+      listOf(
+          Ingredient("Olive Oil", 5, "ml"),
+          Ingredient("Beef Tenderloin", 50, "g"),
+          Ingredient("White Asparagus", 10, "g"),
+          Ingredient("Corn", 4, "g"),
+          Ingredient("Foie Gras", 100, "g"))
+  private val fewPotentialIngredients =
+      listOf(
+          Ingredient("Carrots", 100, "g"),
+      )
+
+  private val manyPotentialIngredients =
+      listOf(
+          Ingredient("Carrots", 100, "g"),
+          Ingredient("Peas", 100, "g"),
+          Ingredient("Worcestershire Sauce", 15, "ml"),
+          Ingredient("Salt", 5, "g"),
+          Ingredient("Pepper", 5, "g"),
+          Ingredient("Garlic", 1, "clove"),
+      )
+
+  @Test
+  fun topBarDisplayed() {
+    launchIngredientScreenWithTestData(emptyList(), emptyList())
+    ComposeScreen.onComposeScreen<IngredientsTopBar>(composeTestRule) {
+      ingredientTitle {
+        assertIsDisplayed()
+        assertTextEquals("Ingredients")
+      }
+
+      backButton {
+        assertIsDisplayed()
+        assertHasClickAction()
+      }
```

Comment:
test that the back button actually goes back to the camera screen/home screen?

Reply:
yep totally forgot to circle back thank you